Sources/WhisperKit/Core/LogitsFilter.swift

```diff
@@ -46,19 +47,200 @@ public class SuppressBlankFilter: LogitsFiltering {
     }
 }
 
+/// Implementation based on https://github.com/openai/whisper/blob/master/whisper/decoding.py#L441
 @available(macOS 13, iOS 16, watchOS 10, visionOS 1, *)
 public class TimestampRulesFilter: LogitsFiltering {
-    let tokenizer: Tokenizer
+    let transcribeToken: Int
+    let translateToken: Int
+    let noTimestampsToken: Int
+    let timeTokenBegin: Int
+    let endToken: Int
     let sampleBegin: Int
-    let maxInitialTimestamp: Int?
+    let maxInitialTimestampIndex: Int?
+    let isModelMultilingual: Bool
 
-    public init(tokenizer: Tokenizer, sampleBegin: Int) {
-        // TODO: implement
-        fatalError("Not implemented: \(#function)")
+    public init(
```

Comment:
This interface is complex in the same way the SegmentSeeker is, I believe most of these are in the tokenizer object but that would require passing this in.

Reply:
you're right, it's complex. I didn't want to make it dependent on `Tokenizer` so it's decoupled and relatively easier to test. I can change it if you think otherwise

Reply:
No problem, your logic makes sense, we may want a simple object like `SpecialTokens` in the future, and extend tokenizer with it, rather than just adding these index properties themselves as extensions.